console/console-init/ui/src/modules/iot-device/components/AddGateways/AddGateway.tsx

```diff
@@ -6,125 +6,333 @@
 import React, { useState, useEffect } from "react";
 import {
   Title,
-  TextInput,
   FormGroup,
   Form,
   Button,
-  FlexItem,
-  Flex,
   Chip,
   ChipGroup,
   GridItem,
-  Grid
+  Grid,
+  Popover,
+  DropdownPosition
 } from "@patternfly/react-core";
-import { DeviceListAlert } from "modules/iot-device/components";
-import { deviceIDRegExp } from "types/Configs";
+import { InfoCircleIcon } from "@patternfly/react-icons";
+import { DropdownWithToggle, TypeAheadSelect } from "components";
+import "./pf-overrides.css";
+import { gatewayTypeOptions } from "modules/iot-device/utils";
 
 export interface IGatewaysProps {
   header?: string;
-  gateways?: string[];
-  returnGateways?: (gateways: string[]) => void;
+  gatewayDevices?: string[];
+  gatewayGroups?: string[];
+  returnGatewayDevices?: (gateways: string[]) => void;
+  returnGatewayGroups?: (gatewayGroups: string[]) => void;
 }
 
 export const AddGateways: React.FunctionComponent<IGatewaysProps> = ({
   header = "Add permitted devices as gateways to this new device",
-  gateways: gatewayList = [],
-  returnGateways
+  gatewayDevices: deviceList = [],
+  gatewayGroups: groupList = [],
+  returnGatewayDevices,
+  returnGatewayGroups
 }) => {
-  const [inputID, setInputID] = useState<string>();
-  const [isValid, setIsValid] = useState<boolean>(false);
-  const [gateways, setGateways] = useState<string[]>(gatewayList);
+  const [gatewayDevices, setGatewayDevices] = useState<string[]>(deviceList);
+  const [gatewayGroups, setGatewayGroups] = useState<string[]>(groupList);
+  const [inputType, setInputType] = useState<string>("device_id");
+  const [selectedGroups, setSelectedGroups] = useState<any[]>([]);
+  const [selectedIDs, setSelectedIDs] = useState<any[]>([]);
 
   useEffect(() => {
-    returnGateways && returnGateways(gateways);
-  }, [gateways]);
+    returnGatewayDevices && returnGatewayDevices(gatewayDevices);
+  }, [gatewayDevices]);
 
-  const addGateway = () => {
-    if (inputID) {
-      setGateways([...gateways, inputID]);
-      setInputID("");
+  useEffect(() => {
+    returnGatewayGroups && returnGatewayGroups(gatewayGroups);
+  }, [gatewayGroups]);
+
+  const addGateway = async () => {
```

Comment:
We can make this logic more simple. Please try below logic if work for you.

const addGateway =() => {  
    let newSelections;
    if(inputType === "device_id" && selections?.length > 0){
      newSelections=[...new Set(gatewayDevices?.concat(selections))];
      setGatewayDevices(newSelections); 
    }else if(inputType === "gateway_group" && selections?.length > 0){
      newSelections=[...new Set(gatewayDevices?.concat(selections))];
      setGatewayGroups(newSelections);
    }
    //reset selections
    setSelections([]); 
  };